tests/integ/modin/test_apply_snowpark_python_functions.py

```diff
@@ -49,3 +49,21 @@ def test_apply_snowpark_python_function_not_implemented():
         pd.DataFrame({"a": [1, 2, 3]}).apply(sin, axis=1)
     with pytest.raises(NotImplementedError):
         pd.DataFrame({"a": [1, 2, 3]}).apply(sin, args=(1, 2))
+
+
+@sql_count_checker(query_count=1)
+def test_apply_snowflake_cortex_summarize():
+    from snowflake.snowpark.functions import snowflake_cortex_summarize
+
+    content = """pandas on Snowflake lets you run your pandas code in a distributed manner directly on your data in
+    Snowflake. Just by changing the import statement and a few lines of code, you can get the familiar pandas experience
+    you know and love with the scalability and security benefits of Snowflake. With pandas on Snowflake, you can work
+    with much larger datasets and avoid the time and expense of porting your pandas pipelines to other big data
+    frameworks or provisioning large and expensive machines. It runs workloads natively in Snowflake through
+    transpilation to SQL, enabling it to take advantage of parallelization and the data governance and security
+    benefits of Snowflake. pandas on Snowflake is delivered through the Snowpark pandas API as part of the Snowpark
+    Python library, which enables scalable data processing of Python code within the Snowflake platform.
+"""
+    s = pd.Series([content])
+    summary = s.apply(snowflake_cortex_summarize)
+    assert 0 < len(summary) < len(content)
```

Comment:
Isn't `len(summary)` getting the length of the series, rather than `len(summary[0])` to get the length of the content?

Besides that, I assume this length check is to get around the fact that this function is non-deterministic. Can you add a comment if that's the case?

Also, are we certain the length of the summary is always shorter than the length of the input?

Reply:
At least for those examples it does shrink the size a lot.